pkg/agent/controllers/agent/types.go

```diff
@@ -0,0 +1,28 @@
+package agent
+
+import (
+	kubeInformers "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+)
+
+type Controller struct {
+	kubeClientSet     kubernetes.Interface
+	serviceWorkqueue  workqueue.RateLimitingInterface
+	servicesSynced    cache.InformerSynced
+	excludeNamespaces map[string]bool
+}
+
+type LighthouseAgentSpecification struct {
```

Comment:
```suggestion
type AgentSpecification struct {
```

I don't think the `Lighthouse` prefix is necessary since it's already in the import path.